topo/node/cisco/cisco.go

```diff
@@ -148,18 +159,117 @@ func (n *Node) Create(ctx context.Context) error {
 		return fmt.Errorf("failed to create pod for %q: %w", pb.Name, err)
 	}
 	log.Debugf("Pod created:\n%+v\n", sPod)
-	log.Infof("Created XRD node resource %s pod", n.Name())
+	log.Infof("Created Cisco %s node resource %s pod", n.Proto.Model, n.Name())
 	if err := n.CreateService(ctx); err != nil {
 		return err
 	}
-	log.Infof("Created XRD node resource %s services", n.Name())
+	log.Infof("Created Cisco %s node resource %s services", n.Proto.Model, n.Name())
 	return nil
 }
 
-func defaults(pb *tpb.Node) *tpb.Node {
+func constraints(pb *tpb.Node) *tpb.Node {
+	if pb.Constraints == nil {
+		pb.Constraints = map[string]string{}
+	}
+	switch pb.Model {
+	case "8201", "8201-32FH", "8202", "8101-32H", "8102-64H":
+		if pb.Constraints["cpu"] == "" {
+			pb.Constraints["cpu"] = "4"
+		}
+		if pb.Constraints["memory"] == "" {
+			pb.Constraints["memory"] = "12Gi"
+		}
+	default:
+		if pb.Constraints["cpu"] == "" {
+			pb.Constraints["cpu"] = "1"
+		}
+		if pb.Constraints["memory"] == "" {
+			pb.Constraints["memory"] = "2Gi"
+		}
+	}
+	return pb
+}
+
+func getCiscoInterfaceId(pb *tpb.Node, eth string) (string, error) {
+	ciscoInterfacePrefix := ""
+	ethWithIdRegx := regexp.MustCompile(`e(t(h(e(r(n(e(t)*)*)*)*)*)*)\d+`) // check for e|et|eth|....
+	ethRegx := regexp.MustCompile(`e(t(h(e(r(n(e(t)*)*)*)*)*)*)`)          // check for e|et|eth|....
+	if !ethWithIdRegx.MatchString(eth) {
+		return "", fmt.Errorf("interface '%s' is invalid", eth)
+	}
+	if pb.Interfaces[eth].Name == "" {
+		// ethWithIdRegx.MatchString(eth) was successfull, so no need to do extra check here
+		ethId, _ := strconv.Atoi(ethRegx.Split(eth, -1)[1])
+		switch pb.Model {
+		case "8201":
```

Comment:
nit so i would refactor this just a touch

```go
func fmtInt100(eid string) string {
  return fmt.Sprintf("HundredGigE0/0/0/%d", eid)
}

func fmtInt400(eid string) string {
  return fmt.Sprintf("FourHundredGigE0/0/0/%d", eid)
}


if !ethWithIdRegx.MatchString(eth) {
	return "", fmt.Errorf("interface '%s' is invalid", eth)
}
if pb.interface[eth].Name != "" {
  return pb.Interfaces[eth].Name, nil
}
// ethWithIdRegx.MatchString(eth) was successful, so no need to do extra check here
ethId, _ := strconv.Atoi(ethRegx.Split(eth, -1)[1])
eid := ethId - 1 
switch pb.Model {
case "8201":
  switch {
    case eid <= 23:
       return fmtInt400(eid), nil
    case eid <= 35:
       return fmtInt100(eid), nil
    }
   return "", fmt.Errorf("interface id %d can not be mapped to a cisco interface, eth1..eth36 is supported on %s ", ethId, pb.Model)
case "8202":
  switch {
    case eid<= 47:
       return fmtInt100(eid), nil
    case eid <= 59:
       return fmtInt400(eid), nil
    case ethId-1 <= 71:
       return fmtInt100(eid), nil
    }
    return "", fmt.Errorf("interface id %d can not be mapped to a cisco interface, eth1..eth72 is supported on %s ", ethId, pb.Model)
case "8201-32FH":
  if eid <= 31 {
       return fmtInt400(eid), nil
  }
  return "", fmt.Errorf("interface id %d can not be mapped to a cisco interface, eth1..eth32 is supported on %s ", ethId, pb.Model)
case "8101-32H":
  if ethId-1 <= 31 {
    return fmtInt100(eid), nil
  }
  return "", fmt.Errorf("interface id %d can not be mapped to a cisco interface, eth1..eth32 is supported on %s ", ethId, pb.Model)
case "8102-64H":
  if ethId-1 <= 63 {
    return fmtInt100(eid), nil
  }
return "", fmt.Errorf("interface id %d can not be mapped to a cisco interface, eth1..eth64 is supported on %s ", ethId, pb.Model)
default:
  return fmt.Sprintf("GigabitEthernet0/0/0/%d", eid)
}
```

Reply:
Thanks a lot! updated as you suggested.